source/include/shadow.h

```diff
@@ -1048,8 +1048,11 @@ ShadowStatus_t Shadow_MatchTopicString( const char * pTopic,
  * See @ref Shadow_MatchTopicString for documentation of common behavior.
  */
 /* @[declare_shadow_matchtopic] */
-#define Shadow_MatchTopic( pTopic, topicLength, pMessageType, pThingName, pThingNameLength ) \
-    Shadow_MatchTopicString( pTopic, topicLength, pMessageType, pThingName, pThingNameLength, NULL, 0 )
+ShadowStatus_t Shadow_MatchTopic( const char * pTopic,
+                                  uint16_t topicLength,
+                                  ShadowMessageType_t * pMessageType,
+                                  const char ** pThingName,
+                                  uint16_t * pThingNameLength );
```

Comment:
Can the thing name length be of `uint8_t *`  type considering that all other APIs are using that width?

Reply:
No. It can't be that type as the old API is using the uint16_t type.